src/utilities/workflow-queries.ts

```diff
@@ -0,0 +1,85 @@
+import * as vscode from 'vscode';
+
+type SavedQuery = vscode.QuickPickItem & { description: string };
+const WORKFLOW_QUERIES = 'workflowQueries';
+
+export const shouldAddQuery = (query: vscode.QuickPickItem) => {
+    const icon = query.buttons?.[0].iconPath as vscode.ThemeIcon | undefined;
+    return icon?.id === new vscode.ThemeIcon('add').id;
```

Comment:
Wondering if there's a better way to tell if `Add Query` has been selected... Maybe if the `QuickPickItem` had a unique ID like `add-query`?